### DISCUSSION TITLE
How to #set page(columns: 2) without starting a new page?

### DISCUSSION THREAD

**Discussion by arj0019** [typst/typst#1107] [General]
I want to display the document metadata and abstract in single column format, and then switch to two column format for the remainder of the document without starting a new page.

An example of the incorrect functionality is provided below:

```
#let title = [PRIMARY TITLE]
#let subtitle = [SECONDARY TITLE]
#let author = "AUTHOR"
#let date = "DATE"

#set page(paper: "a4", margin: (x: 1.5cm, y: 1.5cm), numbering: "1")
#set text(font: "Fira Mono", size: 12pt)
#set par(justify: true, leading: 0.52em)

#align(center, text(18pt)[*#title*] + text(12pt)[\ #subtitle])
#align(center, text(11pt)[#author])
#align(center, text(11pt)[#date])
#line(length: 100%)

#lorem(60)

#set page(columns: 2)

= BODY 1
#lorem(120)

= BODY 2
#lorem(120)

= BODY 3
#lorem(120)
```

![example-1](https://user-images.githubusercontent.com/92353079/236311944-252e01e9-4a00-4852-8085-0a986f6b20e0.png)
![example-2](https://user-images.githubusercontent.com/92353079/236311940-bfe2f240-533d-49bc-81d8-1c50eec983f7.png)

**Comment by arj0019** [comment]
The following solution works, but is not ideal:

```
#let title = [PRIMARY TITLE]
#let subtitle = [SECONDARY TITLE]
#let author = "AUTHOR"
#let date = "DATE"

#set page(paper: "a4", margin: (x: 1.5cm, y: 1.5cm), numbering: "1")
#set text(font: "Fira Mono", size: 12pt)
#set par(justify: true, leading: 0.52em)

#align(center, text(18pt)[*#title*] + text(12pt)[\ #subtitle])
#align(center, text(11pt)[#author])
#align(center, text(11pt)[#date])
#line(length: 100%)

#lorem(60)

#columns(2)[
  = BODY 1
  #lorem(120)
  
  = BODY 2
  #lorem(120)
  
  = BODY 3
 #lorem(120)
]
```

**Reply by laurmaedje** [reply]
You can simplify this with a show rule: `#show: columns.with(2)` followed by the column's content is equivalent to `columns(2)[the content ...]` Then you don't need the extra indentation here.

**Reply by arj0019** [reply]
I like the one line solution, but the container prevents me from adding pagebreaks later in the document. There is probably a way to end the container, but I have not found it yet.

**Reply by laurmaedje** [reply]
Yes, that's unfortunately a limitation currently. Maybe one or two `colbreak()` works for you? It's of course more manual.
